crates/rattler_package_streaming/src/write.rs

```diff
@@ -238,7 +238,7 @@ fn prepare_header(
     header.set_device_major(0)?;
 
     if let Some(timestamp) = timestamp {
-        header.set_mtime(timestamp.timestamp() as u64);
+        header.set_mtime(timestamp.timestamp().unsigned_abs());
```

Comment:
This is a drive-by change. It's the other lines below that actually make the difference